src/game.cpp

```diff
@@ -1789,10 +1789,83 @@ void game::update_weather()
     }
 }
 
+int game::get_heat_radiation( const tripoint &location, bool direct )
+{
+    // Direct heat from fire sources
+    // Cache fires to avoid scanning the map around us bp times
+    // Stored as intensity-distance pairs
+    int temp_mod = 0;
+    std::vector<std::pair<int, int>> fires;
+    fires.reserve( 13 * 13 );
+    int best_fire = 0;
+    for( const tripoint &dest : g->m.points_in_radius( location, 6 ) ) {
+        int heat_intensity = 0;
+
+        int ffire = g->m.get_field_strength( dest, fd_fire );
+        if( ffire > 0 ) {
+            heat_intensity = ffire;
+        } else if( g->m.tr_at( dest ).loadid == tr_lava ) {
+            heat_intensity = 3;
+        }
+        if( heat_intensity == 0 || !g->m.sees( location, dest, -1 ) ) {
+            // No heat source here
+            continue;
+        }
+        // Ensure fire_dist >= 1 to avoid divide-by-zero errors.
+        const int fire_dist = std::max( 1, square_dist( dest, location ) );
+        fires.emplace_back( std::make_pair( heat_intensity, fire_dist ) );
+        if( fire_dist <= 1 ) {
+            // Extend limbs/lean over a single adjacent fire to warm up
+            best_fire = std::max( best_fire, heat_intensity );
+        }
+    }
+
+    for( const auto &intensity_dist : fires ) {
+        const int intensity = intensity_dist.first;
+        const int distance = intensity_dist.second;
+        temp_mod += 6 * intensity * intensity / distance;
+    }
+    if( direct ) {
+        return best_fire;
+    }
+    return temp_mod;
+}
+
+int game::get_convection_temperature( const tripoint &location )
+{
+    // Heat from hot air (fields)
+    int temp_mod = 0;
+    const trap &trap_at_pos = g->m.tr_at( location );
```

Comment:
Whether function is moved from `class game` to be a free function, or remains as is, can shorten all instances of `g->m.__func__` to just `m.__func__`
Other lines impacted:
* 1840
* 1846

Function is only called currently as `g->get_convection_temperature` so it already has access to the global-g's map.

Reply:
If it's freed from game class it losses access to `m` which is member of this class so `g->m` within free functions have to stay.

Reply:
If a map-reference is added as a parameter to the free function you can get rid of the `g->` part. You are correct that if the parameter isn't added `g->m` is the only way you will keep access to the map.